comfy_extras/nodes_string.py

```diff
@@ -0,0 +1,331 @@
+import re
+
+from comfy.comfy_types.node_typing import IO
+
+def normalize_path(path):
+    return path.replace('\\', '/')
+
+class StringConcatenate():
+    @classmethod
+    def INPUT_TYPES(s):
+        return {
+            "required": {
+                "string_a": (IO.STRING, {"multiline": True}),
+                "string_b": (IO.STRING, {"multiline": True})
+            },
+            "optional": {
+                "separator": (IO.STRING, {"default": ""})
+            }
+        }
+
+    RETURN_TYPES = (IO.STRING,)
+    FUNCTION = "execute"
+    CATEGORY = "utils/string"
+
+    def execute(self, string_a, string_b, separator="", **kwargs):
+        return string_a + separator + string_b,
+
+class StringSubstring():
+    @classmethod
+    def INPUT_TYPES(s):
+        return {
+            "required": {
+                "string": (IO.STRING, {"multiline": True}),
+                "start": (IO.INT, {}),
+                "end": (IO.INT, {}),
+            }
+        }
+
+    RETURN_TYPES = (IO.STRING,)
+    FUNCTION = "execute"
+    CATEGORY = "utils/string"
+
+    def execute(self, string, start, end, **kwargs):
+        return string[start:end],
+
+class StringLength():
+    @classmethod
+    def INPUT_TYPES(s):
+        return {
+            "required": {
+                "string": (IO.STRING, {"multiline": True})
+            }
+        }
+
+    RETURN_TYPES = (IO.INT,)
+    RETURN_NAMES = ("length",)
+    FUNCTION = "execute"
+    CATEGORY = "utils/string"
+
+    def execute(self, string, **kwargs):
+        length = len(string)
+
+        return length,
+
+class CaseConverter():
+    @classmethod
+    def INPUT_TYPES(s):
+        return {
+            "required": {
+                "string": (IO.STRING, {"multiline": True}),
+                "mode": (["UPPERCASE", "lowercase", "Capitalize", "Title Case"], {})
```

Comment:
Can you convert the mode syntax to the preferred combo syntax described in https://github.com/comfyanonymous/ComfyUI/blob/16417b40d9411c6e3a63949aa0f3582be25b28db/comfy/comfy_types/node_typing.py#L166-L172

Reply:
done